arangod/RocksDBEngine/RocksDBChecksumEnv.cpp

```diff
@@ -0,0 +1,234 @@
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+#include "RocksDBChecksumEnv.h"
+
+#include "Basics/Exceptions.h"
+#include "Basics/files.h"
+#include "Basics/FileUtils.h"
+#include "Basics/MutexLocker.h"
+#include "Logger/LogMacros.h"
+
+namespace arangodb::checksum {
+
+ChecksumCalculator::ChecksumCalculator()
+    :
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+      _context(EVP_MD_CTX_new()) {
+#else
+      _context(EVP_MD_CTX_create()) {
+#endif
+  if (_context == nullptr) {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_OUT_OF_MEMORY);
+  }
+  if (EVP_DigestInit_ex(_context, EVP_sha256(), nullptr) == 0) {
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL,
+                                   "unable to initialize SHA256 processor");
+  }
+}
+
+ChecksumCalculator::~ChecksumCalculator() {
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+  EVP_MD_CTX_free(_context);
+#else
+  EVP_MD_CTX_destroy(_context);
+#endif
+}
+
+void ChecksumCalculator::update(char const* buffer, size_t n) {
+  TRI_ASSERT(_context != nullptr);
+  EVP_DigestUpdate(_context, static_cast<void const*>(buffer), n);
+}
+
+bool ChecksumHelper::isFileNameSst(std::string const& fileName) noexcept {
```

Comment:
This cannot be `noexcept`, because `TRI_Basename(...)` returns a new std::string by value. That can throw.
```suggestion
bool ChecksumHelper::isFileNameSst(std::string const& fileName) {
```